src/components/Event.tsx

```diff
@@ -1,35 +1,44 @@
-import { useContext, useEffect, useState } from "react";
+import React, { useCallback, useContext, useEffect, useState } from "react";
 import { TicketPurchasingContext } from "../contexts/ticketPurchasing";
 
 export function Event() {
-  const { event, updateEvent } = useContext(TicketPurchasingContext);
+  const { events, selectedEvent, updateEvents, updateSelectedEvent } =
+    useContext(TicketPurchasingContext);
   const [error, setError] = useState(null);
-  const [events, setEvents] = useState([]);
 
   useEffect(() => {
     fetch(`http://localhost:8080/events`)
       .then((res) => res.json())
       .then(
         (result) => {
           console.log(result);
-          setEvents(result);
+          updateEvents(result);
         },
         (error) => {
```

Comment:
ezzel egyet ertesz? olvastam hogy ha nem a then-ben fogadod az error-t akkor miss-olhatsz ui hibakat, te mit szolsz? en eddig mind a catch-ban fogadtam

Reply:
Jol olvastad, rosszul ertetted, es rosszul implementaltad. A [Promise API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch) szerint:

`then(onSuccess, onError)` ekvivalens `then(onSuccess).catch(onError)` statement-el.

A catch-es megoldas szerintem sokkal olvashatobb, nagyon ritkan lattam az elsot. Azzal hogy te a then-be tetted mindkettot nem oldottad meg a problemat amirol olvastal.

A problema le van irva [itt](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful). Ahhoz hogy oldd meg azt a problemat, meg kell nezzed a response.ok vagy response.status-t. A meets-ben ez meg van oldva:

```
fetch('/valami').then(respose => {
  if(!response.ok) {
    throw response; // belemegy a catch-be
  }
  return respnose;
} catch (e) {
  // deal with it
}
```